test/plugin.test.js

```diff
@@ -0,0 +1,464 @@
+'use strict'
+
+const MachinomyServerPlugin = require('..')
+const HDWalletProvider = require('truffle-hdwallet-provider')
+const Store = require('./util/memStore')
+const BtpPacket = require('btp-packet')
+const IlpPacket = require('ilp-packet')
+const debug = require('debug')('ilp-plugin-eohereum-asym-server:test')
+const getPort = require('get-port')
+const formatAmount = require('../build/index.js').formatAmount
+const crypto = require('crypto')
+
+const secret = 'lazy glass net matter square melt fun diary network bean play deer'
+const providerUrl = "https://ropsten.infura.io/T1S8a0bkyrGD7jxJBgeH"
+const provider = new HDWalletProvider(secret, providerUrl, 0)
+function sha256 (preimage) { return crypto.createHash('sha256').update(preimage).digest() }
+
+class Trace extends Error {}
+
+/* TODO Mock for web3, HDWallet, and Machinomy API. */
+async function createPlugin() {
+  /* Silence console log output from provider creation. */
+  const _pre = console.log
+  console.log = () => {}
+
+  let plugin = new MachinomyServerPlugin({
+    address: await provider.getAddress(),
+    provider: provider,
+    minimumChannelAmount: 100,
+    maxPacketAmount: 100,
+    _store: new Store(),
+    port: await getPort(),
+    debugHostIldcpInfo: {
+      clientAddress: 'test.prefix',
+      assetScale: 9,
+      assetCode: 'ETH'
+    }
+  })
+
+  console.log = _pre
+
+  return plugin
+}
+
+
+beforeEach(async () => {
+  try {
+    this.plugin = await createPlugin() 
```

Comment:
I'm not 100% sure how this works; `this` shouldn't be bound to anything in this context because it's an arrow function. I guess it's using the `this` of the global scope? Either way, functions in the tests should be changed to `function` instead of arrow function syntax.

Reply:
The reason I did this is because Jest does not share the 'this' context across beforeEach, afterEach, and the test itself (as mocha does). 

https://github.com/facebook/jest/issues/3553

To get the shared variable access, I either had to use arrow functions with 'this', or define  variables as globals (as indicated above). I chose to use 'this' because I thought it would be visually more clear which variables were acting as globals. Can change it to the latter if it is best to have each test function individually scoped.

Reply:
Ah, I see. I guess I'm just not as familiar with jest. If this is the common way that it's done then that's fine